bindings/python/pymongocrypt/credentials.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from collections import namedtuple
+from datetime import datetime, timedelta
+
+try:
+    from pymongo_auth_aws.auth import aws_temp_credentials
+    _HAVE_AUTH_AWS = True
+except ImportError:
+    _HAVE_AUTH_AWS = False
+
+from pymongocrypt.errors import MongoCryptError
+
+import requests
+
+
+_azure_creds = namedtuple("_azure_creds", ["access_token", "expires_in"])
+_azure_creds_cache = None
```

Comment:
Why do we cache azure creds but not GCP creds? Is that coming later?

Reply:
In the scope document for this issue it was determined that caching was not needed for GCP creds.